include/git2/global.h

```diff
@@ -38,6 +38,28 @@ GIT_EXTERN(int) git_libgit2_init(void);
  */
 GIT_EXTERN(int) git_libgit2_shutdown(void);
 
+#if defined(GIT_WIN32) && defined(_INC_WINDOWS)
```

Comment:
If it was declared in "features.h.in", then this here could become `#if defined(GIT_WIN32) && defined(GIT_STATIC)`.

Reply:
The test for _INC_WINDOWS is absolutely necessary because we don't know at which point the user application has included Windows.h or whether the user application wants to include Windows.h at all.
Otherwise we must include Windows.h from git2.h so that the windows data types for the DLL handler prototype are available.

Reply:
This is probably not what you want.

Reply:
Fair enough :)

Reply:
You may have a point though. I am not quite sure for which use case this thread handling via DllMain() is meant. Does the HTTPS or any other Transport mode create threads internally in libgit2, or is it for when a user application uses libgit2 in ist own multiple threads?

If it is the former, we really should let the compiler bail out so the user knows he must do something (call git_libgit2_dllmain() for static links) or it breaks libgit2.